src/town_gui.cpp

```diff
@@ -993,7 +993,7 @@ struct TownDirectoryWindow : public Window {
 					this->towns.Sort();
 					this->towns.shrink_to_fit();
 					this->towns.RebuildDone();
-					this->vscroll->SetCount(this->towns.size()); // Update scrollbar as well.
+					this->vscroll->SetCount((int)this->towns.size()); // Update scrollbar as well.
```

Comment:
Do negative numbers have any sort of special meaning for SetCount? Or would the "proper" solution be to make the signature take an unsigned number?

Reply:
Asserts in `SetCount()` enforce the value to be compatible with `uint16`. So maybe scrollbars need to be "improved"

Reply:
Probably needs a rethink, but that's something for later